lib/acts_as_taggable_on/taggable/related.rb

```diff
@@ -49,7 +49,7 @@ def related_tags_for(context, klass, options = {})
     private
 
     def exclude_self(klass, id)
-      "#{klass.table_name}.#{klass.primary_key} != #{id} AND" if [self.class.base_class, self.class].include? klass
+      "#{klass.table_name}.#{klass.primary_key} != '#{id}' AND" if [self.class.base_class, self.class].include? klass
```

Comment:
This fixed uuid PK but broke integer/bigint

Reply:
@seuros Thank you for pointing out my mistake.

How do you think if I use `klass.columns_hash['id'].type` to check the PK type first, and modify the method like this:

```rb
def exclude_self(klass, id)
  new_id = klass.columns_hash['id'].type == :uuid ? "'#{id}'" : id

  "#{klass.table_name}.#{klass.primary_key} != #{new_id} AND" if [self.class.base_class, self.class].include? klass
end
```

Reply:
I think we should use Arel in this part of code.

Reply:
@seuros Thanks for your suggestion.
I'm not familiar with Arel, would you please confirm the following code is correct.

```rb
def exclude_self(klass, id)
  "#{klass.arel_table[klass.primary_key].not_eq(id).to_sql} AND" if [self.class.base_class, self.class].include? klass
end
```

Reply:
Yes, i think it should work.
Can you rebase your PR ?

Reply:
Also please include a test case.